qiskit/dagcircuit/dagnode.py

```diff
@@ -74,7 +74,7 @@ def semantic_eq(node1, node2, bit_indices1=None, bit_indices2=None):
             node2_cargs = [bit_indices2[carg] for carg in node2.cargs]
 
             # For barriers, qarg order is not significant so compare as sets
-            if "barrier" == node1.op.name == node2.op.name:
+            if node1.op.name == node2.op.name and node1.name in {"barrier", "swap"}:
```

Comment:
What about cz? But I feel like this should be an operation property that the qarg order doesn't matter because it comes up in other places too like `GateDirection` (right now we hard code cz, but swap will have the same problem)

Reply:
`cz` used to be in this list as well in an earlier iteration of this PR - I don't know why it got dropped. Was there a reason, @ewinston?